apps/web/src/components/Basenames/UsernameProfileSectionBadges/Badges/index.tsx

```diff
@@ -24,59 +25,157 @@ import baseInitiateGray from './images/baseInitiateGray.png';
+  },
+  BUILDATHON_WINNER: {
+    name: 'Buildathon Winner',
+    title: 'Buildathon Winner',
+    description: 'You submitted a winning project in the Onchain Summer 2024 Buildathon. Congrats!',
+    cta: 'Learn more',
+    ctaLink: 'https://www.base.org/onchainsummer',
+    image: buildathonWinner,
+    grayImage: buildathonWinnerGray,
+  },
+  TALENT_SCORE: {
+    name: 'Talent Passport Score',
+    title: 'Talent Passport Builder Score',
+    description: "Your builder score as a Talent passport holder. You're legit!",
+    cta: 'Get your talent passport',
+    ctaLink: 'https://passport.talentprotocol.com/',
+    image: talentScore,
+    grayImage: talentScoreGray,
+  },
 };
 
-export const GRAY_BADGE_IMGS: Record<BadgeNames, StaticImport> = {
-  VERIFIED_IDENTITY: verifiedIdentityGray as StaticImport,
-  VERIFIED_COUNTRY: verifiedCountryGray as StaticImport,
-  VERIFIED_COINBASE_ONE: verifiedCoinbaseOneGray as StaticImport,
-  BASE_BUILDER: baseBuilderGray as StaticImport,
-  BASE_GRANTEE: baseGranteeGray as StaticImport,
-  BASE_INITIATE: baseInitiateGray as StaticImport,
-  BASE_LEARN_NEWCOMER: baseLearnNewcomerGray as StaticImport,
-  BUILDATHON_PARTICIPANT: buildathonParticipantGray as StaticImport,
-  BUILDATHON_WINNER: buildathonWinnerGray as StaticImport,
-};
-
-export const BADGE_NAMES: Record<BadgeNames, string> = {
-  VERIFIED_IDENTITY: 'Coinbase Verified ID',
-  VERIFIED_COUNTRY: 'Verified Country',
-  VERIFIED_COINBASE_ONE: 'Coinbase One',
-  BASE_BUILDER: 'Based Builder',
-  BASE_GRANTEE: 'Base Grantee',
-  BASE_INITIATE: 'Based Initiate',
-  BASE_LEARN_NEWCOMER: 'Base Learn Newcomer',
-  BUILDATHON_PARTICIPANT: 'Buildathon Participant',
-  BUILDATHON_WINNER: 'Buildathon Winner',
-};
+export function Badge({
+  badge,
+  claimed,
+  score,
+}: {
+  badge: BadgeNames;
+  claimed?: boolean;
+  score?: number;
+}) {
+  const { selectBadge } = useBadgeContext();
+  const onClick = useCallback(() => {
+    selectBadge({ badge, claimed: !!claimed });
+  }, [selectBadge, badge, claimed]);
 
-export function Badge({ badge, earned }: { badge: BadgeNames; earned?: boolean }) {
-  const name = BADGE_NAMES[badge];
+  const name = BADGE_INFO[badge].name;
   return (
     <div className="flex flex-col gap-4">
-      <div className="relative flex h-[160px] w-[160px] items-center justify-center rounded-[24px] border border-gray-10">
+      <div
+        className="relative flex h-[160px] w-[160px] items-center justify-center rounded-[24px] border border-gray-10"
+        onClick={onClick}
+        onKeyDown={onClick}
+        aria-label={`See details for ${name}`}
+        role="button"
+        tabIndex={0}
+      >
         <Image
-          src={(earned ? BADGE_IMGS : GRAY_BADGE_IMGS)[badge]}
+          src={BADGE_INFO[badge][claimed ? 'image' : 'grayImage']}
           alt={name}
           height={100}
           width={100}
         />
-        {!earned && (
+        {claimed && score && badge === 'TALENT_SCORE' && (
```

Comment:
nit: compound `&&` used to render templates can sometimes abort early and if one of the leading values is only truthy you get unexpected strings/numbers rendered. I like to avoid it and assign these to a descriptive variable like this:

`const showTalentScore = Boolean(claimed && score && badge === 'TALENT_SCORE')`

Reply:
That's a good one